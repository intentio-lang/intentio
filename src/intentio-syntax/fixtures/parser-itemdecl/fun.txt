fun foo(a, b) {
    a;
}
