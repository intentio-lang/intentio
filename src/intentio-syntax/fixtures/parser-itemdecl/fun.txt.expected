[ FunDecl 
    { _itemDeclName = ScopeId "foo"
    , _funDeclParams = FunParams 
        [ FunParam ( ScopeId "a" )
        , FunParam ( ScopeId "b" )
        ] 
    , _funDeclBody = FunBody ( Block [ IdExpr ( Id ( ScopeId "a" ) ) ] )
    } 
, FunDecl 
    { _itemDeclName = ScopeId "id"
    , _funDeclParams = FunParams [ FunParam ( ScopeId "x" ) ]
    , _funDeclBody = FunBody ( Block [ IdExpr ( Id ( ScopeId "x" ) ) ] )
    } 
, FunDecl 
    { _itemDeclName = ScopeId "constant"
    , _funDeclParams = FunParams []
    , _funDeclBody = FunBody ( Block [ LitExpr ( Integer "92" ) ] )
    } 
] 